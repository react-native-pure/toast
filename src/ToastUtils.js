import * as React from "react"
import type {ToastOption} from "./Types";
import Sibling from "@react-native-pure/sibling"
import Toast from "./Toast"

export default {
    show: (option: ToastOption) => {
        const remove = Sibling.append(
            <Toast {...option} remove={remove}/>
        );
    }
}