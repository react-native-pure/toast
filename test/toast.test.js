import * as React from "react"
import renderer from 'react-test-renderer'
import WrapperComponent from "@react-native-pure/sibling/src/WrapperComponent"
import ToastUtils from "../src/ToastUtils"
import {Text, View} from "react-native"

test(`show toast`, (done) => {
    const onShown = jest.fn(() => {
        expect(onShown.mock.calls.length).toBe(1);
    });
    const onHidden = jest.fn(() => {
        expect(onHidden.mock.calls.length).toBe(1);
    });
    const component = renderer.create(
        <WrapperComponent></WrapperComponent>
    );
    ToastUtils.show({
        message: "message",
        onShown,
        onHidden
    });
    expect(component.toJSON()).toMatchSnapshot();
    setTimeout(() => {
        expect(component.toJSON()).toMatchSnapshot();
        done();
    }, 5 * 1000);
});

test(`show center toast`, (done) => {
    const component = renderer.create(
        <WrapperComponent></WrapperComponent>
    );
    ToastUtils.show({
        message: "message",
        position: "center"
    });
    expect(component.toJSON()).toMatchSnapshot();
    setTimeout(done, 5 * 1000);
});

test(`show bottom toast`, (done) => {
    const component = renderer.create(
        <WrapperComponent></WrapperComponent>
    );
    ToastUtils.show({
        message: "message",
        position: "bottom"
    });
    expect(component.toJSON()).toMatchSnapshot();
    setTimeout(done, 5 * 1000);
});

test(`show toast during 1s`, (done) => {
    const component = renderer.create(
        <WrapperComponent></WrapperComponent>
    );
    ToastUtils.show({
        message: "message",
        duration: 1000
    });
    expect(component.toJSON()).toMatchSnapshot();
    setTimeout(done, 5 * 1000);
});

test(`disable onPress`, (done) => {
    const component = renderer.create(
        <WrapperComponent></WrapperComponent>
    );
    ToastUtils.show({
        message: "message",
        hideOnPress: false
    });
    expect(component.toJSON()).toMatchSnapshot();
    setTimeout(done, 5 * 1000);
});

test(`custom renderMessage`, (done) => {
    const component = renderer.create(
        <WrapperComponent></WrapperComponent>
    );
    ToastUtils.show({
        message: "abc",
        renderMessage: (message, style) => {
            return (
                <View style={{borderRadius: 4}}>
                    <Text style={style}>{message}</Text>
                </View>
            )
        }
    });

    expect(component.toJSON()).toMatchSnapshot();
    setTimeout(done, 5 * 1000);
});

test(`message is Object`, (done) => {
    const component = renderer.create(
        <WrapperComponent></WrapperComponent>
    );
    ToastUtils.show({
        message: {
            m1: "hello",
            m2: "world"
        },
        renderMessage: ({m1, m2}, style) => {
            return (
                <View style={{borderRadius: 4}}>
                    <Text style={style}>{m1},{m2}</Text>
                </View>
            )
        }
    });

    expect(component.toJSON()).toMatchSnapshot();
    setTimeout(done, 5 * 1000);
})

