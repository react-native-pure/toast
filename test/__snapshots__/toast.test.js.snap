// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom renderMessage 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
        },
        null,
        Object {
          "top": 20,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "borderRadius": 4,
        }
      }
    >
      <Text
        style={null}
      >
        abc
      </Text>
    </View>
  </View>
</View>
`;

exports[`disable onPress 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
        },
        null,
        Object {
          "top": 20,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "backgroundColor": "black",
            "color": "white",
          },
          null,
        ]
      }
    >
      message
    </Text>
  </View>
</View>
`;

exports[`message is Object 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
        },
        null,
        Object {
          "top": 20,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "borderRadius": 4,
        }
      }
    >
      <Text
        style={null}
      >
        hello
        ,
        world
      </Text>
    </View>
  </View>
</View>
`;

exports[`show bottom toast 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
        },
        null,
        Object {
          "bottom": 20,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "backgroundColor": "black",
            "color": "white",
          },
          null,
        ]
      }
    >
      message
    </Text>
  </View>
</View>
`;

exports[`show center toast 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
        },
        null,
        Object {
          "top": "50%",
          "transfrom": Array [
            Object {
              "translateY": "-50%",
            },
          ],
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "backgroundColor": "black",
            "color": "white",
          },
          null,
        ]
      }
    >
      message
    </Text>
  </View>
</View>
`;

exports[`show icon toast 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
        },
        null,
        Object {
          "top": 20,
        },
      ]
    }
  >
    <Text>
      Icon
    </Text>
    <Text
      style={
        Array [
          Object {
            "backgroundColor": "black",
            "color": "white",
          },
          null,
        ]
      }
    >
      message
    </Text>
  </View>
</View>
`;

exports[`show icon toast that disable onPress 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
        },
        null,
        Object {
          "top": 20,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "backgroundColor": "black",
            "color": "white",
          },
          null,
        ]
      }
    >
      message
    </Text>
  </View>
</View>
`;

exports[`show toast 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
        },
        null,
        Object {
          "top": 20,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "backgroundColor": "black",
            "color": "white",
          },
          null,
        ]
      }
    >
      message
    </Text>
  </View>
</View>
`;

exports[`show toast 2`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;

exports[`show toast during 1s 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
        },
        null,
        Object {
          "top": 20,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "backgroundColor": "black",
            "color": "white",
          },
          null,
        ]
      }
    >
      message
    </Text>
  </View>
</View>
`;
